OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/notifications/internal/permissions/impl/NotificationPermissionController.kt

```diff
@@ -64,14 +75,57 @@ internal class NotificationPermissionController(
             )!!
 
     init {
+        this.enabled = notificationsEnabled()
         _requestPermission.registerAsCallback(PERMISSION_TYPE, this)
+        pollingWaitInterval = _configModelStore.model.foregroundFetchNotificationPermissionInterval
```

Comment:
We should start with the background time here to be safe. I believe the onFcous event fires right away if the app is already in focus so it will be updated right away if the app happens to really be in the foreground at this point.

Reply:
Good catch! Yes it does fire immediately and wakes so it starts polling correctly